EOS-13891 beck: enabled directio for stob domain

jamohite opened the pull request:
With previous code directio=false, the value of adom->sad_babshift was getting 
incorrectly derived hence there was issue in reserving balloc extents.
Now by default directio=true, and If option -I is provided in beck, directio will be disabled.

Signed-off-by: Jaydeep Mohite <jaydeep.mohite@seagate.com>

cortx-admin commented:
<h1>Jenkins CI Result : <a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62/>Motr#62</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>2</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/47motr-tests-user-kernel target=_blank>47motr-tests-user-kernel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/32m0t1fs-rconfc-fail-test target=_blank>32m0t1fs-rconfc-fail-test</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>5</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/31motr-sys-kvs-kernel-test target=_blank>31motr-sys-kvs-kernel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/07multi-client-tests target=_blank>07multi-client-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/39mount-fail target=_blank>39mount-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/12sns-cc-io-tests target=_blank>12sns-cc-io-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/37protocol target=_blank>37protocol</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>55</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/40motr-dgmode target=_blank>40motr-dgmode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/51motr-rmw target=_blank>51motr-rmw</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/50motr-rm-lock-cc-io target=_blank>50motr-rm-lock-cc-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/20pool-version-test target=_blank>20pool-version-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/46m0crate target=_blank>46m0crate</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/54sns-repair-motr-1f target=_blank>54sns-repair-motr-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/25spiel-sns-repair-test target=_blank>25spiel-sns-repair-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/21spiel-test target=_blank>21spiel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/59sns-repair-motr-abort target=_blank>59sns-repair-motr-abort</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/01kernel-tests target=_blank>01kernel-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/33motr-st target=_blank>33motr-st</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/42motr-utils target=_blank>42motr-utils</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/19sns-abort-test target=_blank>19sns-abort-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/09poolmach-tests target=_blank>09poolmach-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/13sns-repair-quiesce target=_blank>13sns-repair-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/41motr-conf-update target=_blank>41motr-conf-update</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/17sss-st-test target=_blank>17sss-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/00userspace-tests target=_blank>00userspace-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/03initscript-tests target=_blank>03initscript-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/45motr-sns-repair-N-1 target=_blank>45motr-sns-repair-N-1</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/30hagen-test target=_blank>30hagen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/11sns-multi-tests target=_blank>11sns-multi-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/58sns-repair-motr-ios-fail target=_blank>58sns-repair-motr-ios-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/29confgen-test target=_blank>29confgen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/34sns-repair-1n-1f target=_blank>34sns-repair-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/44motr-sns-repair target=_blank>44motr-sns-repair</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/08degraded-mode-tests target=_blank>08degraded-mode-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/26spiel-sns-repair-quiesce-test target=_blank>26spiel-sns-repair-quiesce-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/60sns-repair-motr-abort-quiesce target=_blank>60sns-repair-motr-abort-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/35m0singlenode target=_blank>35m0singlenode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/14m0t1fs-fsync-test target=_blank>14m0t1fs-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/43motr-sync-replication target=_blank>43motr-sync-replication</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/04net-tests target=_blank>04net-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/55sns-repair-motr-1n-1f target=_blank>55sns-repair-motr-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/24m0d-fsync-test target=_blank>24m0d-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/49motr-rpc-cancel target=_blank>49motr-rpc-cancel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/56sns-repair-motr-mf target=_blank>56sns-repair-motr-mf</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/38sns-abort-quiesce-test target=_blank>38sns-abort-quiesce-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/15m0t1fs-fwait-test target=_blank>15m0t1fs-fwait-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/02system-tests target=_blank>02system-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/22rpc-cancel-test target=_blank>22rpc-cancel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/53clusterusage-alert target=_blank>53clusterusage-alert</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/05rpcping-tests target=_blank>05rpcping-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/23m0d-signal-test target=_blank>23m0d-signal-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/28motr-sys-kvs-test target=_blank>28motr-sys-kvs-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/06console-tests target=_blank>06console-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/18m0mount-test target=_blank>18m0mount-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/27sns-repair-io-fail-test target=_blank>27sns-repair-io-fail-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/16conf-st-test target=_blank>16conf-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/52singlenode-sanity target=_blank>52singlenode-sanity</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/57sns-repair-motr-1k-1f target=_blank>57sns-repair-motr-1k-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/48motr-raid0-io target=_blank>48motr-raid0-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/10sns-single-tests target=_blank>10sns-single-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/36spare-reservation target=_blank>36spare-reservation</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node/51kem target=_blank>51kem</a></p></details></td></tr><tr><td>Total</td><td>62</td><td><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/62//testReport/junit/motr-single-node>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

max-seagate reviewed (changes requested):
> EOS-13891 beck: Updated for directio flag during stob domain init and added -I option to disable directio #99

What does "updated" mean here? Which stob domain? Please make the commit message summary line short and unabiguous.

jamohite commented:
> > EOS-13891 beck: Updated for directio flag during stob domain init and added -I option to disable directio #99
> 
> What does "updated" mean here? Which stob domain? Please make the commit message summary line short and unabiguous.

Updated directio=true from directio=false.
